common/lib/dependabot/config/ignore_condition.rb

```diff
@@ -4,37 +4,40 @@ module Dependabot
   module Config
     # Filters versions that should not be considered for dependency updates
     class IgnoreCondition
-      UPDATE_TYPES = %i(
-        ignore_major_versions
-        ignore_minor_versions
-        ignore_patch_versions
-      ).freeze
+      PATCH_VERSION_TYPE = "version-update:semver-patch"
+      MINOR_VERSION_TYPE = "version-update:semver-minor"
+      MAJOR_VERSION_TYPE = "version-update:semver-major"
 
       ALL_VERSIONS = ">= 0"
 
       attr_reader :dependency_name, :versions, :update_types
+
       def initialize(dependency_name:, versions: nil, update_types: nil)
         @dependency_name = dependency_name
         @versions = versions || []
         @update_types = update_types || []
       end
 
       def ignored_versions(dependency)
-        return [ALL_VERSIONS] if @versions.empty? && @update_types.empty?
+        return [ALL_VERSIONS] if versions.empty? && transformed_update_types.empty?
 
-        versions_by_type(dependency) + @versions
+        versions_by_type(dependency) + versions
       end
 
       private
 
+      def transformed_update_types
+        update_types.map(&:downcase).map(&:strip).compact
```

Comment:
```suggestion
        update_types.compact.map(&:downcase).map(&:strip)
```

Nit: this `compact` does nothing, as nil values have already raised?
```
irb(main):002:0> [nil].map(&:downcase)
(irb):2:in `map': undefined method `downcase' for nil:NilClass (NoMethodError)
```

I'm not sure if it should be refactored to be a guard for the `map()`s (the suggestion), or just removed

Reply:
Question: should this transform happen in the `ConfigFile` / parsing logic?

We might be able to simplify by assuming the caller of `IgnoreCondition` has normalized these values: either through `ConfigFile` or the mysterious server-side parsing process.

Reply:
Good catch! Yeah true, might be better to move to the `File` class and rely on the api to have done this already